### PR TITLE
Buck publish to support zip_file

### PR DESCRIPTION
**Pull request by linzhp**
This PR adds support to `buck publish` command to publish zip archive to Maven repositories.

# why do we need this
We need to 1) publish home brew bottles; 2) upload third-party Go packages to Artifactory so we can build a project even when the original repository of a third-party package is deleted. Both need to publish zip archive.

# why existing publish functionality doesn't work
Existing `buck publish` command require the target has `maven` flavor, which zip_file targets doesn't support. In fact zip_file doesn't need to build differently to be published to Maven

# brief overview changes in the code.
* Add `maven_coords` parameter to `zip_file`
* Stop adding flavors to `zip_file` targets

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -208,6 +208,7 @@
     <include name="aether/aether-transport-http-1.0.2.v20150114.jar" />
     <include name="aether/aether-transport-file-1.0.2.v20150114.jar" />
     <include name="aether/aether-util-1.0.2.v20150114.jar" />
+    <include name="commons-codec/commons-codec-1.6.jar" />
```
Comment (kageiit) [inline build.xml]: Why do we need this change?

Comment (linzhp) [reply build.xml]: Because of a NoClassDef error

**Comment by linzhp** [comment]
@ttsugriy Could you take a look?

**Comment by styurin** [comment]
Could you provide in the description:

- why do we need this
- why existing publish functionality doesn't work
- brief overview changes in the code.

**Comment by linzhp** [comment]
Updated description. @styurin

**Comment by v-jizhang** [comment]
Sorry we never got back to this! Thank you for your contributions. If you are still interested in adding zip_file support, please rebase and update this PR and we promise to look at it in a timely manner. Thank you (and sorry) again.

**Comment by v-jizhang** [comment]
Closing this PR because there is no update for 8 months.

**Pull request closed without merging**